aws/resource_aws_ssm_parameter.go

```diff
@@ -47,45 +52,88 @@ func resourceAwsSsmParameter() *schema.Resource {
 			"key_id": {
 				Type:     schema.TypeString,
 				Optional: true,
-				ForceNew: true,
 			},
 			"overwrite": {
 				Type:     schema.TypeBool,
 				Optional: true,
-				Default:  false,
 			},
+			"allowed_pattern": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "",
+			},
+			"tags": tagsSchema(),
 		},
 	}
 }
 
+func resourceAwsSmmParameterExists(d *schema.ResourceData, meta interface{}) (bool, error) {
+	ssmconn := meta.(*AWSClient).ssmconn
+
+	resp, err := ssmconn.GetParameters(&ssm.GetParametersInput{
+		Names:          []*string{aws.String(d.Id())},
+		WithDecryption: aws.Bool(true),
+	})
+
+	if err != nil {
+		return false, err
+	}
+	return len(resp.InvalidParameters) == 0, nil
+}
+
 func resourceAwsSsmParameterRead(d *schema.ResourceData, meta interface{}) error {
 	ssmconn := meta.(*AWSClient).ssmconn
 
 	log.Printf("[DEBUG] Reading SSM Parameter: %s", d.Id())
 
-	paramInput := &ssm.GetParametersInput{
-		Names: []*string{
-			aws.String(d.Id()),
-		},
+	resp, err := ssmconn.GetParameters(&ssm.GetParametersInput{
+		Names:          []*string{aws.String(d.Id())},
 		WithDecryption: aws.Bool(true),
+	})
+	if err != nil {
+		return errwrap.Wrapf("[ERROR] Error getting SSM parameter: {{err}}", err)
 	}
 
-	resp, err := ssmconn.GetParameters(paramInput)
+	param := resp.Parameters[0]
+	d.Set("name", param.Name)
+	d.Set("type", param.Type)
+	d.Set("value", param.Value)
 
+	respDetailed, err := ssmconn.DescribeParameters(&ssm.DescribeParametersInput{
+		Filters: []*ssm.ParametersFilter{
+			&ssm.ParametersFilter{
+				Key:    aws.String("Name"),
+				Values: []*string{aws.String(d.Get("name").(string))},
+			},
+		},
+	})
 	if err != nil {
 		return errwrap.Wrapf("[ERROR] Error describing SSM parameter: {{err}}", err)
 	}
 
-	if len(resp.Parameters) == 0 {
```

Comment:
Is there a reason why this logic was removed? We should force recreation of the parameter if its missing or at least prevent a Terraform crash by ensuring we won't get an index out of bounds error if `resp.Parameters` has no elements.